sql-parser/src/main/java/io/crate/sql/tree/CreateFunction.java

```diff
@@ -0,0 +1,144 @@
+
+import com.google.common.base.MoreObjects;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class CreateFunction extends Statement {
+
+    public enum FunctionLanguage {
+        JAVASCRIPT
+    }
+
+    public enum Option {
+        STRICT("STRICT"),
+        CALLED_ON_NULL_INPUT("CALLED ON NULL INPUT"),
+        RETURNS_NULL_ON_NULL_INPUT("RETURNS NULL ON NULL INPUT");
+
+        private final String option;
+
+        Option(String option) {
+            this.option = option;
+        }
+
+        @Override
+        public String toString() {
+            return option;
+        }
+    }
+
+    private final QualifiedName name;
+    private final boolean replace;
+    private final List<FunctionArgument> arguments;
+    private final ColumnType returnType;
+    private final FunctionLanguage language;
+    private final String body;
+    private final Set<Option> options;
+
+    public CreateFunction(QualifiedName name,
+                          boolean replace,
+                          List<FunctionArgument> arguments,
+                          ColumnType returnType,
+                          Set<Option> options,
+                          FunctionLanguage language,
+                          String body) {
+        this.name = name;
+        this.replace = replace;
+        this.arguments = arguments;
+        this.returnType = returnType;
+        this.language = language;
+        this.body = body;
+        this.options = options;
+    }
+
+    public QualifiedName name() {
+        return name;
+    }
+
+    public boolean replace() {
+        return replace;
+    }
+
+    public List<FunctionArgument> arguments() {
+        return arguments;
+    }
+
+    public ColumnType returnType() {
+        return returnType;
+    }
+
+    public FunctionLanguage getLanguage() {
+        return language;
+    }
+
+    public String body() {
+        return body;
+    }
+
+    public Set<Option> options() {
+        return options;
+    }
+
+    @Override
+    public <R, C> R accept(AstVisitor<R, C> visitor, C context) {
+        return visitor.visitCreateFunction(this, context);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (getClass() != o.getClass()) return false;
+
+        final CreateFunction that = (CreateFunction) o;
+        return Objects.equals(this.name, that.name)
+            && Objects.equals(this.replace, that.replace)
+            && Objects.equals(this.arguments, that.arguments)
+            && Objects.equals(this.returnType, that.returnType)
+            && Objects.equals(this.language, that.language)
+            && Objects.equals(this.options, that.options);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, replace, arguments, returnType, language, options);
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
```

Comment:
For the future: I'd prefer that we use the other toString template that doesn't depend on guava